sdk/dynamicconfig/sdk/metric/controller/push/push.go

```diff
@@ -253,13 +253,16 @@ func (c *Controller) tick() {
 		// Export all metrics with the same period at the same time.
 		overdue := []int32{}
 		now := c.clock.Now()
+		// Have a tolerance of 10% of the period
+		tolerance := c.period / time.Duration(10)
 
 		for period, lastCollect := range c.lastCollected {
 			expectedExportTime := lastCollect.Add(time.Duration(period) * time.Second)
+			expectedExportTimeWithTolerance := expectedExportTime.Add(-tolerance)
```

Comment:
Just a small suggestion to simplify:

```suggestion
		tolerance := c.period / time.Duration(10)

		for period, lastCollect := range c.lastCollected {
			expectedExportTimeWithTolerance := lastCollect.Add(time.Duration(period) * time.Second - tolerance)
```

Reply:
Looks good!